Sources/MapboxMaps/Style/Style.swift

```diff
@@ -132,21 +84,54 @@ public class Style {
     public func _layer(with layerID: String, type: Layer.Type) -> Result<Layer, LayerError> {
 
         // Get the layer properties from the map
-        let layerProps = styleManager.getStyleLayerProperties(forLayerId: layerID)
-
-        guard layerProps.isValue(),
-              let validValue = layerProps.value as? [String: AnyObject] else {
-            return .failure(.getStyleLayerFailed(layerProps.error as? String))
-        }
-        // Decode the layer properties into a layer object
         do {
-            let layer = try type.init(jsonObject: validValue)
+            let layerProps = try layerProperties(for: layerID)
+            let layer = try type.init(jsonObject: layerProps)
             return .success(layer)
         } catch {
             return .failure(.layerDecodingFailed(error))
         }
     }
 
+    /// Updates a layer that exists in the style already
+    /// - Parameters:
+    ///   - id: identifier of layer to update
+    ///   - type: Type of the layer
+    ///   - update: Closure that mutates a layer passed to it
+    /// - Returns: Result type with  `.success` if update is successful, `LayerError` otherwise
+    @discardableResult
+    public func updateLayer<T: Layer>(id: String, type: T.Type, update: (inout T) -> Void) -> Result<Bool, LayerError> {
```

Comment:
Will this be updated as part of a future PR?

Reply:
Yep.